.github/workflows/pr_ci.yml

```diff
@@ -31,6 +31,12 @@ jobs:
       # warnings and other better approaches don't work, see https://github.com/OrchardCMS/OrchardCore/pull/16317.
       run: |
         dotnet build -c Release -warnaserror /p:TreatWarningsAsErrors=true /p:RunAnalyzers=true /p:NuGetAudit=false
+    - name: Restore dependencies
+      run: dotnet restore
+    - name: Set Git line ending configuration
+      run: git config --global core.autocrlf true
+    - name: Validate code format
+      run: dotnet format --verify-no-changes --verbosity diagnostic
```

Comment:
This shouldn't be necessary, due to the build causing warnings to fail.

Reply:
without this, there is no guarantee that the formatting is correct,

Reply:
Analyzers, including code style analyzers, integrate into the build too, causing build warnings/errors, so can fail the build without `dotnet format` (you only need that if you want to apply formatting, or verify formatting without a build).

Reply:
The idea behind `dotnet format` is to run it and see if there is changes "just like we do with assets validation".

Reply:
Unlike asset validation, we don't need it though :). We need the build to take care of this, because then it's one less moving part, and easier to reproduce the violations in any environment, and it'll be supported by any IDE.

This is not some kind of hack BTW, analyzers are meant to run as part of the build, this is a standard, recommended approach. It's also necessary for all analyzers not used by `dotnet format`.

Reply:
I fixed the build behavior. Having to double-declare IDE* rules is also something we painstakingly learned with https://github.com/Lombiq/.NET-Analyzers.

Please do any clean-up necessary (like I suppose none of the currently commented-out code in the CI workflow is needed) and make sure that `OrchardCore.Commons.props` is the right file to put these into (I'm not sure; it looks like it, but the hierarchy of these props files, and how they reference each other, is complex).

Reply:
Thanks! Everything now looks find and should be where needs to be. the CI is working as expected now.